### PR TITLE
`Development`: Add JavaDocs to UtilServices and Factories [5]

### PR DESCRIPTION
**Pull request by DominikRemo**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

Improve the readability of UtilServices and Factories by adding JavaDoc to the helper functions

### Description

Added documentation to following files:
- [x] exercise/fileuploadexercise/FileUploadExerciseFactory.java
- [x] exercise/fileuploadexercise/FileUploadExerciseUtilService.java

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

**Review by laadvo — COMMENTED** [review]
Already looks good, just one thing you should consider addressing

File: src/test/java/de/tum/in/www1/artemis/exercise/fileuploadexercise/FileUploadExerciseUtilService.java [hunk 1]
```diff
@@ -125,11 +150,23 @@ public List<FileUploadExercise> createFourFileUploadExercisesWithCourseWithCusto
...
     }
 
+    /**
+     * Saves the given FileUploadExercises. The given List must contain exactly four FileUploadExercises belonging to the same Course. The course must not contain more than four
+     * exercises.
+     *
+     * @param fileUploadExercises The List of FileUploadExercises to save. Must contain exactly four FileUploadExercises belonging to the same Course.
+     * @return The Course the FileUploadExercises belong to.
+     */
     private Course addFileUploadExercisesToCourse(List<FileUploadExercise> fileUploadExercises) {
```
[the remark this answers is not in the document]
Comment (DominikRemo) [reply src/test/java/de/tum/in/www1/artemis/exercise/fileuploadexercise/FileUploadExerciseUtilService.java]: Removed the method

**Review by chrisknedl — COMMENTED** [review]
Looks good, only one thing: I noticed you sometimes use full stops for some of the `@param` sections, and sometimes not. Is there a reason for this?

**Comment by DominikRemo** [comment]
> Looks good, only one thing: I noticed you sometimes use full stops for some of the `@param` sections, and sometimes not. Is there a reason for this?

@chrisknedl Thank you for looking through this PR. There was no reason for the fullstops and I have removed them to be more consistent

**Review by chrisknedl — APPROVED** [review]
thanks for implementing my suggestion! looks perfect now 👍

**Pull request merged**